packages/trader/src/AppV2/Components/TradeParameters/Barrier/barrier-input.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { ActionSheet, Chip, Text, TextField, TextFieldAddon } from '@deriv-com/quill-ui';
+
+import { localize, Localize } from '@deriv/translations';
+import { observer } from 'mobx-react';
+import { useTraderStore } from 'Stores/useTraderStores';
+
+const chips_options = [
+    {
+        name: <Localize i18n_default_text='Above spot' />,
+    },
+    {
+        name: <Localize i18n_default_text='Below spot' />,
+    },
+    {
+        name: <Localize i18n_default_text='Fixed price' />,
+    },
+];
+const BarrierInput = observer(
+    ({ setInitialBarrierValue, isDays }: { setInitialBarrierValue: (val: string) => void; isDays: boolean }) => {
+        const { barrier_1, onChange, validation_errors, proposal_info } = useTraderStore();
+        const [option, setOption] = React.useState(0);
+
+        const proposal = Object.values(proposal_info);
+        const spotPrice = proposal.length > 0 ? proposal[0].spot : '';
+
+        React.useEffect(() => {
+            setInitialBarrierValue(barrier_1);
+            if (barrier_1.includes('-')) {
+                setOption(1);
+            } else if (barrier_1.includes('+')) {
+                setOption(0);
+            } else {
+                setOption(2);
+            }
+        }, [setInitialBarrierValue]);
```

Comment:
Do we need this function in the array of dependencies? useEffect will be triggered each time reference to function change